www/include/views/graphs/graph-split.php

```diff
@@ -48,174 +49,169 @@
 $tpl = new Smarty();
 $tpl = initSmartyTpl($path, $tpl);
 
-function getGetPostValue($str)
-{
-    $value = null;
-    if (isset($_GET[$str]) &&
-        $_GET[$str]
-    ) {
-        $value = $_GET[$str];
-    }
-    if (isset($_POST[$str]) &&
-        $_POST[$str]
-    ) {
-        $value = $_POST[$str];
-    }
-    return urldecode($value);
-}
+$chartId = filter_var($_GET['chartId'] ?? null, FILTER_SANITIZE_STRING);
 
-$svc_id = getGetPostValue('chartId');
+if (preg_match('/([0-9]+)_([0-9]+)/', $chartId, $matches)) {
+    $hostId = (int)$matches[1];
+    $serviceId = (int)$matches[2];
+} else {
+    throw new \InvalidArgumentException('chartId must be a combination of integers');
+}
 
 $metrics = array();
-if (isset($svc_id) && $svc_id) {
-    list($hostId, $svcId) = explode('_', $svc_id);
-    /* Get list metrics */
-    $query = 'SELECT m.metric_id, m.metric_name, i.host_name, i.service_description ' .
-        'FROM metrics m, index_data i ' .
-        'WHERE i.id = m.index_id AND i.service_id = ' . CentreonDB::escape($svcId) .
-        ' AND i.host_id = ' . CentreonDB::escape($hostId);
-    $res = $pearDBO->query($query);
-    while ($row = $res->fetch()) {
-        $metrics[] = array(
-            'id' => $svc_id . '_' . $row['metric_id'],
-            'title' => $row['host_name'] . ' - ' . $row['service_description'] . ' : ' . $row['metric_name']
-        );
-    }
+
+/* Get list metrics */
+$query = 'SELECT m.metric_id, m.metric_name, i.host_name, i.service_description
+    FROM metrics m, index_data i
+    WHERE i.id = m.index_id AND i.service_id = :serviceId
+    AND i.host_id = :hostId';
+
+$stmt = $pearDBO->prepare($query);
+$stmt->bindValue(':serviceId', $serviceId, \PDO::PARAM_INT);
+$stmt->bindValue(':hostId', $hostId, \PDO::PARAM_INT);
+$stmt->execute();
+
+while ($row = $stmt->fetch(\PDO::FETCH_ASSOC)) {
+    $metrics[] = array(
+        'id' => $chartId . '_' . $row['metric_id'],
+        'title' => $row['host_name'] . ' - ' . $row['service_description'] . ' : ' . $row['metric_name']
+    );
 }
 
-/* Get Period if is in url */
-$period_start = 'undefined';
-$period_end = 'undefined';
-if (isset($_REQUEST['start']) &&
-    is_numeric($_REQUEST['start'])
-) {
-    $period_start = $_REQUEST['start'];
+if (isset($_GET['start'])) {
+    $period_start = filter_var($_GET['start'], FILTER_VALIDATE_INT);
 }
```

Comment:
As, if $_GET['start'] is not provided in the request, you'll fail the check line 195
https://github.com/centreon/centreon/pull/9738/files#diff-86135526cd024f1a05fa4e102aebd3b448d025713936b844c4476cbe7543554cR195

Same TODO for end_date
```suggestion
if (isset($_GET['start'])) {
    $period_start = filter_var($_GET['start'], FILTER_VALIDATE_INT) ?? 'undefined';
} else {
    $period_start = 'undefined';
}
```